src/utils/body.ts

```diff
@@ -18,20 +19,37 @@ export type ParseBodyOptions = {
    * parseBody should return { file: ['aaa', 'bbb'], message: 'hello' }
    */
   all?: boolean
+  /**
+   * Determines whether all fields with dot notation should be parsed as nested objects.
+   * @default false
+   * @example
+   * const data = new FormData()
+   * data.append('obj.key1', 'value1')
+   * data.append('obj.key2', 'value2')
+   *
+   * If dot is false:
+   * parseBody should return { 'obj.key1': 'value1', 'obj.key2': 'value2' }
+   *
+   * If dot is true:
+   * parseBody should return { obj: { key1: 'value1', key2: 'value2' } }
+   */
+  dot?: boolean
 }
 
 export const parseBody = async <T extends BodyData = BodyData>(
   request: HonoRequest | Request,
-  options: ParseBodyOptions = { all: false }
+  options: ParseBodyOptions = Object.create(null)
```

Comment:
We can change this to be `options: Partial<ParseBodyOptions> = Object.create(null)`